cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -97,97 +100,82 @@ public void removeApplication(String applicationName) {
      * @throws NullPointerException     when bean or app name is null
      *                                  or docker password was not set in environment variables when creating app via
      *                                  dockerImage and docker credentials
-     * @throws IllegalArgumentException when neither a path nor a docker image were specified, or app name empty
-     * @throws CreationException        when app already exists
-     *                                  or any fatal error occurs during creation of the app
+     * @throws IllegalArgumentException    when neither a path nor a docker image were specified, or app name empty
+     * @throws CreationException        when any fatal error occurs during creation of the app
      * @throws SecurityException        when there is no permission to access environment variable CF_DOCKER_PASSWORD
+     * @return Mono which can be subscribed on to trigger the creation of the app
      */
-    public void create(String appName, ApplicationBean bean, boolean shouldStart) throws CreationException {
+    public Mono<Void> create(String appName, ApplicationBean bean, boolean shouldStart) {
         checkNotNull(appName);
         checkArgument(!appName.isEmpty(), "empty name");
         checkNotNull(bean);
 
-        // useful, otherwise cloud foundry operations library might behave in a weird way
-        // path null + docker image null => NullPointer Exception that is not intuitive
-        // and when setting docker image to empty string to prevent this
-        // can lead to clash when path and buildpack was set
-        checkIfPathOrDockerGiven(bean);
-
-        // this check is important, otherwise an app could get overwritten
-        if (appExists(appName)) {
-            throw new CreationException("app exists already");
-        }
-
         try {
-            Log.debug("Create app:", appName);
-            Log.debug("Bean of the app:", bean);
-            Log.debug("Should the app start:", shouldStart);
-
-            doCreate(appName, bean, shouldStart);
+            return doCreate(appName, bean, shouldStart);
         } catch (RuntimeException e) {
```

Comment:
Is it still possible, that a RuntimeException occurs in doCreate?

Reply:
Yep, still possible, for example when neither path or docker image was given.